wrapper/src/main/java/software/amazon/jdbc/util/CacheMap.java

```diff
@@ -21,51 +21,106 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
 
 public class CacheMap<K, V> {
 
-  private final Map<K, CacheItem<V>> cache = new ConcurrentHashMap<>();
-  private final long cleanupIntervalNanos = TimeUnit.MINUTES.toNanos(10);
+  private final Map<K, CacheItem> cache = new ConcurrentHashMap<>();
+  private long cleanupIntervalNanos = TimeUnit.MINUTES.toNanos(10);
```

Comment:
Should this be configurable via the HikariPooledConnectionProvider constructor? Right now there isn't a way for the user to set this